### PR TITLE
Automatically format source code with pre-commit

### PR DESCRIPTION
**Pull request by komamitsu**
ScalarDB developers execute `gradlew spotlessApply` or IDE code formatter manually as a part of development. But the manual operation should be automated and documented for new contributors. This PR describes how to setup the automated operation. The drawback is every commit invokes `gradlew spotless` and it would increase the latency of commit a bit (especially for the first invocation of Gradle daemon.)

### CODE REVIEW

**Review by brfrn169 — COMMENTED** [review]
LGTM other than what I commented. Please take a look when you have time!

File: README.md [hunk 1]
```diff
@@ -62,6 +62,20 @@ Here are the contributors we are especially thankful for:
 - [Toshihiro Suzuki](https://github.com/brfrn169) - created [Phoenix adapter](https://github.com/scalar-labs/scalardb-phoenix) for ScalarDB
 - [Yonezawa-T2](https://github.com/Yonezawa-T2) - reported bugs around Serializable and proposed a new Serializable strategy (now named Extra-Read)
 
+## Development
```
[the remark this answers is not in the document]
Comment (komamitsu) [reply README.md:65]: Done https://github.com/scalar-labs/scalardb/pull/812/commits/7d8cfacf1e47244ed7ce5e71d719640c6694e1fb

**Review by Torch3333 — APPROVED** [review]
LGTM, thank you!

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Review by brfrn169 — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**